nextcord/cooldowns/cooldown.py

```diff
@@ -0,0 +1,253 @@
+# The MIT License (MIT)
+# Copyright (c) 2021-present vcokltfre & tag-epic
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+
+from __future__ import annotations
+
+import asyncio
+import functools
+import time
+from asyncio import Future
+from asyncio.events import AbstractEventLoop, get_event_loop
+from logging import getLogger
+from typing import TYPE_CHECKING, Callable, Optional, TypeVar
+
+from nextcord.cooldowns import CooldownBucket
+from nextcord.cooldowns.buckets import _HashableArguments
+from nextcord.cooldowns.protocols import Bucket
```

Comment:
this should be named BucketProtocol instead

Reply:
don’t we have other protocols which don’t have protocol in the name? Unless your talking about a as import

Reply:
Probably, however they are hopefully going to be renamed otherwise imports are kinda shadowed